.github/workflows/self_hosted_runner_utils/setup_runner.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# Copyright 2022 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# More or less copied from
+# https://github.com/iree-org/iree/tree/main/build_tools/github_actions/runner/config
+
+set -ex
+
+if [ "$#" -ne 3 ]; then
+  echo "Usage: setup_runner.sh <runner name> <tags> <github token>"
+fi
+
+runner_name=$1
+runner_tags=$2
+runner_token=$3
+
+# Secret fourth argument for setting the repo URL. Useful for testing with forks.
+jax_repo_url=$4
+if [ -z "${jax_repo_url}" ]; then
+  jax_repo_url="https://github.com/google/jax"
+fi
+
+# Create `runner` user. This user won't have sudo access unless you ssh into the
+# GCP VM as `runner` using gcloud. Don't do that!
+sudo useradd runner -m
+
+# Find the latest actions-runner download
```

Comment:
Are you sure you want automatic updates here and not a pinned version?

Reply:
Yeah if you turn off auto-update you have to pick up the new version within 30 days. I turned off auto-update and we manually manage bumping the pinned version. This is more work, but something something supply chain security (plus random breakages that happen at confusing times)

Reply:
Oh, basically the thing it says in the ratchet README :-)

Reply:
Hrrrrmph I'm gonna risk it, I think the chances of us not updating within 30 days is high, especially since it looks like they update quite often? https://github.com/actions/runner/releases

Reply:
Yeah fair. It is a bit annoying to remember to update